[feat/notification] notification serializer 추가

han-son-03 opened the pull request:
## ✅ PR 요약
- 관련 이슈 번호: #123
- 작업 요약: notification serializer 추가

## 📄 상세 내용
- [x] 알림 특성상 대부분 일방향 통신이므로 read_only_fields를 지정하여 정보 신뢰성을 유지
- [x] NotificationReadSerializer를 만들어, 사용자가 단건 혹은 a전체 알림 조회 시 is_read 상태를 안전하게 처리할 수 있도록 고려


## 📸 스크린샷 (선택)

## 📝 기타 참고 사항
apps/notification/__init__.py가 없어서 추가했습니다.

## 🧪 PR Checklist
- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다.(버그 수정/기능에 대한 테스트).

han-son-03 committed:
✨ feat: notification serializer 추가

알림 특성상 대부분 일방향 통신이므로 read_only_fields를 지정하여 정보 신뢰성을 유지

NotificationReadSerializer를 만들어, 사용자가 단건 혹은 전체 알림 조회 시 is_read 상태를 안전하게 처리할 수 있도록 고려함

2joonkim commented on apps/notification/serializers/notification_serializers.py:
```diff
@@ -0,0 +1,26 @@
+class NotificationReadSerializer(serializers.ModelSerializer["Notification"]):
+    class Meta:
+        model = Notification
+        fields = ["is_read"]
```
이렇게 된다면  is_read를 False로 다시 변경하는 것이 허용됩니다
읽음처리만 가능하게 하려면 validate_is_read 메서드를 추가하여 True만 받도록 검증하는 것이 좋겠죠?

han-son-03 replied:
여기도 추가하겠습니다

2joonkim commented on apps/notification/serializers/notification_serializers.py:
```diff
@@ -0,0 +1,26 @@
+        model = Notification
+        fields = [
+            "id",
+            "user",
```
필드에 유저는 왜 들어가있을까요?!

han-son-03 replied:
수정하겠습니다!

han-son-03 committed:
✨ feat: NotificationReadSerializer is_read 검증 추가

NotificationReadSerializer에서 is_read 필드가 True만 허용되도록 하여 False나 다른 값으로 변경하는 시도 차단

han-son-03 committed:
🎨 style    : test.sh실행!

테스트 ci 마지막줄 공백 때문에 터졌습니다.
